tests/unit/stanmath/StanPerceptronTest_unittest.cc

```diff
@@ -0,0 +1,73 @@
+#include <gtest/gtest.h>
+#include <stan/math.hpp>
+#include <iostream>
+#include <cmath>
+#include <Eigen/Dense>
+#include <random>
+using namespace std;
+using namespace Eigen;
+using namespace stan::math;
+
+
+/*
+This test will implement a perceptron using stan math
+The perceptron will have one input, the column vector [1 1 1]^T
+Then, it will have 2x3 Matrix of weights that it will learn
+Then, the output is a 2x1 column vector
+In this example, we want to learn the weights W such that the square
+Error loss from the output of the perceptron to [1 1]^T is minimized.
+Since we can find weights from [1 1 1]^T to [1 1]^T in a perceptron,
+this error should be very close to zero after 100 epochs.
+*/
+TEST(StanPerceptronTest, sample_perceptron)
```

Comment:
while I like this test, I think we should have something super simple. Like the gradient of the gaussian kernel wrt to bandwidth, left/right argument. Something that we know in closed form.

Reply:
But please de-prioritise the test until all the cmake stuff is correct

Reply:
this test is very fine, lets concentrate on the actual inclusion of the lib